provisioning/device/src/Authentication/AuthenticationProviderSymmetricKey.cs

```diff
@@ -16,6 +18,8 @@ public class AuthenticationProviderSymmetricKey : AuthenticationProvider
         /// <param name="registrationId">The Provisioning service Registration Id for this device.</param>
         /// <param name="primaryKey">The primary key for this device.</param>
         /// <param name="secondaryKey">The secondary key for this device.</param>
+        /// <exception cref="ArgumentException">When the required parameter is an empty string or consists only of white-space characters.</exception>
+        /// <exception cref="ArgumentNullException">When the required parameter is null.</exception>
```

Comment:
```suggestion
        /// <exception cref="ArgumentNullException">Either <paramref name="registrationId"/>,  <paramref name="registrationId"/>, or <paramref name="secondaryKey"/> is null.</exception>

Reply:
I would rather keep stating "required parameters" than those specific param names since we are doing exactly same checks on all of 3 params, plus, we can save some space.